src/generic/MPoly.jl

```diff
@@ -1,2847 +1,2989 @@
-         end
-         if v.i != 0 || v.j < m
-            push!(Q, x.n)
-         else
-            push!(reuse, x.n)
-         end
-         while (xn = v.next) != 0
-            v = I[xn]
-            if v.i == 0
-               qc = addmul!(qc, a.coeffs[m + 1 - v.j], m1, c)
-            else
-               qc = addmul!(qc, b[v.p].coeffs[n[v.p] + 1 - v.i], Qc[v.p][v.j], c)
-            end
-            if v.i != 0 || v.j < m
-               push!(Q, xn)
-            else
-               push!(reuse, xn)
-            end
-         end
-      end
-      @inbounds while !isempty(Q)
-         xn = pop!(Q)
-         v = I[xn]
-         if v.i == 0
-            I[xn] = nheap_t(0, v.j + 1, 0, 0)
-            vw = Viewn[viewc]
-            monomial_sub!(Exps, vw, maxn, 1, a.exps, m - v.j, N)
-            if nheapinsert!(H, I, xn, vw, Exps, N, 0) # either chain or insert into heap  
-               viewc -= 1
-            end 
-         elseif v.j < k[v.p]
-            I[xn] = nheap_t(v.i, v.j + 1, v.p, 0)
-            vw = Viewn[viewc]
-            monomial_sub!(temp, 1, maxn, 1, b[v.p].exps, n[v.p] + 1 - v.i, N)
-            monomial_sub!(Exps, vw, temp, 1, Qe[v.p], v.j + 1, N)
-            if nheapinsert!(H, I, xn, vw, Exps, N, v.p) # either chain or insert into heap
-               viewc -= 1
-            end
-         elseif v.j == k[v.p]
-            s[v.p] += 1
-            push!(reuse, xn)
-         end  
-      end
-      if qc != 0
-         div_flag = false
-         for w = 1:len
-            monomial_sub!(temp, 1, maxn, 1, b[w].exps, n[w], N)
-            d1 = monomial_divides!(texp, 1, temp, 1, exp_copy, 1, mask, N)
-            if d1
-               d2, tq = divides(qc, mb[w])
-               if d2
-                  div_flag = true
-                  k[w] += 1
-                  if k[w] > q_alloc[w]
-                     q_alloc[w] *= 2
-                     resize!(Qc[w], q_alloc[w])
-                     tt = reshape(Qe[w], N*size(Qe[w], 2))
-                     resize!(tt, N*q_alloc[w])
-                     Qe[w] = reshape(tt, N, q_alloc[w])
-                  end
-                  Qc[w][k[w]] = tq
-                  monomial_set!(Qe[w], k[w], texp, 1, N)
-                  for i = 2:s[w]
-                     if !isempty(reuse)
-                        xn = pop!(reuse)
-                        I[xn] = nheap_t(i, k[w], w, 0)
-                        vw = Viewn[viewc]
-                        monomial_sub!(temp, 1, maxn, 1, b[w].exps, n[w] + 1 - i, N)
-                        monomial_sub!(Exps, vw, temp, 1, Qe[w], k[w], N)
-                        if nheapinsert!(H, I, xn, vw, Exps, N, w) # either chain or insert into heap
-                           viewc -= 1
-                        end
-                     else
-                        push!(I, nheap_t(i, k[w], w, 0))
-                        vw = Viewn[viewc]
-                        monomial_sub!(Exps, vw, maxn, 1, b[w].exps, n[w] + 1 - i, N)
-                   
```

Comment:
Wrong line, sorry. There is one call to `isdefined(b, 1)` somewhere. If you change it to `isassigned(b, 1)`, most of the deprecations warnings are gone and tests won't time out.